compose/cli/docker_client.py

```diff
@@ -32,4 +32,4 @@ def docker_client():
         )
 
     timeout = int(os.environ.get('DOCKER_CLIENT_TIMEOUT', 60))
-    return Client(base_url=base_url, tls=tls_config, version='1.18', timeout=timeout)
+    return Client(base_url=base_url, tls=tls_config, version='1.19', timeout=timeout)
```

Comment:
`auto` is cool but introduces another HTTP request to every invocation of `docker-compose`, which I don't think is an acceptable slowdown.

As for making users upgrade: yeah, it sucks, but once Compose 1.4 is released, Docker 1.6 will be two versions old. Part of the reason we do lockstep releases of the whole platform is so that we don't have to cater to older versions (although we've been good about that so far).

I'm OK with adding the environment variable if it defaults to `1.19` and we tell Docker 1.6 users that they can set it back if they want but it won't be officially supported.

@gheart Your variable name is more descriptive, but inventing new `DOCKER_` variables that don't exist in the Docker client is not Compose's job, which is why we prefix ours with `COMPOSE_`.

Since you're eager to get a contribution in, if you rename the variable to `COMPOSE_API_VERSION`, and update your PR (squashed to one commit), I'd be happy to merge it. I'll then rebase this PR to update the default version to `1.19`.